rfcs/0052-extensible-udt/0052-extensible-udt.md

```diff
@@ -0,0 +1,370 @@
+---
+Number: "0052"
+Category: Standards Track
+Status: Proposal
+Author: Xuejie Xiao <xxuejie@gmail.com>, Xu Jiandong <lynndon@gmail.com>
+Created: 2024-01-09
+---
+
+
+# Extensible UDT
+
+Extensible UDT(xUDT) is an extension of [Simple
+UDT](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0025-simple-udt/0025-simple-udt.md) for
+defining more behaviors a UDT might need. While simple UDT provides a minimal
+core for issuing UDTs on Nervos CKB, extensible UDT builds on top of simple UDT
+for more potential needs, such as regulations.
+
+## **Data Structure**
+
+**xUDT Cell**
+
+An xUDT cell is backward compatible with Simple UDT, all the existing rules
+defined in the Simple UDT spec must still hold true for xUDT cells. On top of
+sUDT, xUDT extends a cell like the following:
+
+```yaml
+data:
+    <amount: uint128> <xUDT data>
+type:
+    code_hash: xUDT type script
+    args: <owner lock script hash> <xUDT args>
```

Comment:
```suggestion
    code_hash: xUDT type script
    hash_type: data1 on mainnet | type on testnet
    args: <owner lock script hash> <xUDT args>
```

In the current deployments, developers using xUDT will have to switch the `hash_type` of xUDT type_script between testnet and mainnet.

Reply:
In spec, we can't write that. We can deploy this script on local devnet with other code_hash/hash_type.